tdrs-backend/tdpservice/data_files/serializers.py

```diff
@@ -39,8 +40,28 @@ class Meta:
 
     def create(self, validated_data):
         """Create a new entry with a new version number."""
-        return DataFile.create_new_version(validated_data)
+        data_file = DataFile.create_new_version(validated_data)
+
+        # Determine the matching ClamAVFileScan for this DataFile.
+        av_scan = ClamAVFileScan.objects.filter(
+            file_name=data_file.original_filename,
+            uploaded_by=data_file.user
+        ).last()
+
+        # Link the newly created DataFile to the relevant ClamAVFileScan.
+        if av_scan is not None:
+            av_scan.data_file = data_file
+            av_scan.save()
+
+        return data_file
 
     def update(self, instance, validated_data):
         """Throw an error if a user tries to update a data_file."""
         raise ImmutabilityError(instance, validated_data)
+
+    def validate_file(self, file):
+        """Perform all validation steps on a given file."""
+        user = self.context.get('user')
```

Comment:
This context needs to be passed in by the implementing view(s) in the `get_serializer_context` method.